packages/notification/src/features/inProgress/handler.ts

```diff
@@ -0,0 +1,165 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * OpenCRVS is also distributed under the terms of the Civil Registration
+ * & Healthcare Disclaimer located at http://opencrvs.org/license.
+ *
+ * Copyright (C) The OpenCRVS Authors located at https://github.com/opencrvs/opencrvs-core/blob/master/AUTHORS.
+ */
+import { Request, ResponseToolkit } from '@hapi/hapi'
+import { getDefaultLanguage } from '@notification/i18n/utils'
+import {
+  findCompositionSection,
+  findExtension,
+  findResourceFromBundleById,
+  getComposition,
+  getResourceFromBundleById,
+  getStatusFromTask,
+  getTaskFromSavedBundle,
+  getTrackingId,
+  InProgressRecord,
+  Location,
+  Patient,
+  RelatedPerson,
+  resourceIdentifierToUUID,
+  SavedRelatedPerson,
+  urlReferenceToUUID
+} from '@opencrvs/commons/types'
+import { sendNotification } from '@notification/features/sms/utils'
+import { messageKeys } from '@notification/i18n/messages'
+
+function error(record: InProgressRecord, message: string): never {
+  const task = getTaskFromSavedBundle(record)
+  const taskStatus = getStatusFromTask(task)
+  throw new Error(`${message} in ${taskStatus} record`)
+}
+
+function getOfficeName(record: InProgressRecord) {
+  const task = getTaskFromSavedBundle(record)
+  const officeExtension = findExtension(
+    'http://opencrvs.org/specs/extension/regLastOffice',
+    task.extension
+  )
+  return officeExtension?.valueString
+}
+
+function getInformantName(record: InProgressRecord) {
+  const composition = getComposition(record)
+  const informantSection = findCompositionSection(
+    'informant-details',
+    composition
+  )
+  if (!informantSection) {
+    return null
+  }
+  const informantRelation: Partial<SavedRelatedPerson> =
+    getResourceFromBundleById<RelatedPerson>(
+      record,
+      urlReferenceToUUID(informantSection.entry[0].reference)
+    )
+  if (!informantRelation.patient?.reference) {
+    return null
+  }
+  const informant = findResourceFromBundleById<Patient>(
+    record,
+    urlReferenceToUUID(informantRelation.patient.reference)
+  )
+  const name = informant?.name?.find(({ use }) => use === 'en')
+  if (!name) {
+    return null
+  }
+  return [name.given?.join(' '), name.family.join(' ')].join(' ').trim()
+}
+
+function getRegistrationLocation(record: InProgressRecord) {
+  const task = getTaskFromSavedBundle(record)
+  const locationExtension = findExtension(
+    'http://opencrvs.org/specs/extension/regLastLocation',
+    task.extension
+  )
+  if (!locationExtension) {
+    error(record, 'no last registration office found')
+  }
```

Comment:
Is this something that has a clear reason why it could happen or is it one of those that should not happen ever?

Reply:
Maybe this is related to my second comment. We could have some sort of 
`new UnexpectedRecordError(record, 'No last registration office found')`

Reply:
In the notification service this might as well be regarded as a 400 Bad Request as it was not expecting the payload to be like this right?